packages/loopring_v3.js/src/types.ts

```diff
@@ -0,0 +1,412 @@
+  tokenA: number,
+  /** The amount of tokens (in tokenS) the taker sells. */
+  fillSA: BN;
+  /** The fee (in tokenB) paid by the taker. */
+  feeA: BN;
+  /** The protocol fee that needs to be paid by the operator for the taker. */
+  protocolFeeA: BN;
+  /** The amount of tokens (in tokenB) the taker receives as rebate. */
+  rebateA: BN;
+
+  /** The account of the maker. */
+  accountIdB: number;
+  /** The orderID of the maker order. */
+  orderIdB: number;
+  /** Whether the maker order is a buy or sell order. */
+  buyB: boolean;
+  /** The token the maker order sells. */
+  tokenB: number,
+  /** The amount of tokens (in tokenS) the maker sells. */
+  fillSB: BN;
+  /** The fee (in tokenB) paid by the maker. */
+  feeB: BN;
+  /** The protocol fee that needs to be paid by the operator for the maker. */
+  protocolFeeB: BN;
+  /** The amount of tokens (in tokenB) the maker receives as rebate. */
+  rebateB: BN;
+}
+
+/**
+ * Off-chain withdrawal data.
+ */
+export interface OffchainWithdrawal {
+  /** The exchange the off-chain withdrawal request was made on. */
+  exchangeId: number;
+  /** The block this off-chain withdrawal request was pocessed in. */
+  blockIdx: number;
+  /** The request index of this off-chain withdrawal in the processed requests list (@see getProcessedRequest). */
+  requestIdx: number;
+
+  /** The account this withdrawal is for. */
+  accountID: number;
+  /** The token that is being withdrawn. */
+  tokenID: number;
+  /** The amount that was actually withdrawn. */
+  amountWithdrawn: BN;
+  /** The token the fee to the operator is paid in. */
+  feeTokenID: number;
+  /** The fee paid to the operator. */
+  fee: BN;
+}
+
+/**
+ * Order cancellation data.
+ */
+export interface OrderCancellation {
+  /** The exchange the order canellation request was made on. */
+  exchangeId: number;
+  /** The block this order cancellation request was pocessed in. */
+  blockIdx: number;
+  /** The request index of this oorder cancellation in the processed requests list (@see getProcessedRequest). */
+  requestIdx: number;
+
+  /** The account this order cancellation is for. */
+  accountID: number;
+  /** The tokenS of the order that is being cancelled. */
+  orderTokenID: number;
+  /** The orderID of the order that is being cancelled. */
+  orderID: number;
+  /** The token the fee to the operator is paid in. */
+  feeTokenID: number;
+  /** The fee paid to the operator. */
+  fee: BN;
+}
+
+/**
+ * Trade history data.
+ */
+export interface TradeHistory {
+  /** How much the order is filled. */
+  filled: BN;
+  /** Wheter the order is cancelled. */
+  cancelled: boolean;
+  /** The orderID of the order the trade history is currently stored for. */
+  orderID: number;
+}
+
+/**
+ * Balance data.
+ */
+export interface Balance {
+  /** How amount of tokens the account owner has for a token. */
+  balance: BN;
+  /** The trade history data. */
+  tradeHistory: { [key: number]: TradeHistory };
```

Comment:
I kind of expected this, but I was unsure how it would actually work best with the database so I left it like this.

I will change it to use separate lists if that's how you want to do it in the database.